notebooks/04. EcoMap & EcoPlot/EcoMap.ipynb

```diff
@@ -276,47 +273,40 @@
    "source": [
     "m = EcoMap(width=800, height=600)\n",
     "\n",
-    "# Add two tiled basemaps (OSM and Google Satellite Hybrid)\n",
-    "m.add_basemap(\"OpenStreetMap\")\n",
-    "m.add_basemap(\"HYBRID\")\n",
+    "# Add tiled base layer\n",
+    "m.add_layer(EcoMap.get_named_tile_layer(\"OpenStreetMap\"))\n",
     "\n",
     "# Set DEM visualization parameters\n",
     "vis_params = {\"min\": 0, \"max\": 4000, \"opacity\": 0.5, \"palette\": [\"006633\", \"E5FFCC\", \"662A00\", \"D8D8D8\", \"F5F5F5\"]}\n",
     "\n",
     "# Add Google Earth Engine elevation layer\n",
     "dem = ee.Image(\"USGS/SRTMGL1_003\")\n",
-    "m.add_ee_layer(dem.updateMask(dem.gt(0)), vis_params, \"DEM\")\n",
-    "\n",
-    "# Zoom in and add regions outlines\n",
-    "m.zoom_to_gdf(region_gdf)\n",
-    "m.add_gdf(\n",
-    "    region_gdf,\n",
-    "    style_kwds={\"fillOpacity\": 0.1, \"opacity\": 0.5, \"color\": \"black\"},\n",
-    "    color=[\"#7fc97f\", \"#beaed4\", \"#fdc086\", \"#ffff99\"],\n",
-    ")\n",
+    "m.add_ee_layer(dem.updateMask(dem.gt(0)), vis_params)\n",
+    "\n",
+    "zone_colors = {\n",
+    "    1: EcoMap.hex_to_rgb(\"#7fc97f\"),\n",
+    "    2: EcoMap.hex_to_rgb(\"#beaed4\"),\n",
+    "    3: EcoMap.hex_to_rgb(\"#fdc086\"),\n",
+    "    4: EcoMap.hex_to_rgb(\"#ffff99\"),\n",
+    "}\n",
+    "cmap = apply_categorical_cmap(region_gdf.ZoneID, zone_colors)\n",
+    "m.add_polygon_layer(region_gdf, opacity=0.5, get_fill_color=cmap, zoom=True)\n",
+    "\n",
     "\n",
     "# Add trajectory\n",
-    "vehicle_gdf.geometry.explore(m=m, color=\"#468af7\")\n",
+    "m.add_path_layer(vehicle_gdf, get_width=200, get_color=EcoMap.hex_to_rgb(\"#468af7\"))\n",
     "\n",
     "# Add elephant sighting events\n",
-    "m.add_gdf(\n",
-    "    events_gdf[[\"geometry\", \"serial_number\", \"location\", \"title\", \"event_type\"]],\n",
-    "    color=\"#f746ad\",\n",
-    "    marker_type=\"circle_marker\",\n",
-    "    marker_kwds={\"radius\": 7, \"fill\": True, \"draggable\": False},\n",
-    ")\n",
+    "m.add_scatterplot_layer(events_gdf, get_radius=700, get_fill_color=EcoMap.hex_to_rgb(\"#f746ad\"))\n",
     "\n",
     "# Add title\n",
-    "m.add_title(title=\"Elephant Sighting Map\", align=\"center\", font_size=\"18px\")\n",
+    "m.add_title(\"Elephant Sighting Map\")\n",
     "\n",
-    "# Add north-arrow. Positions are: topright, topleft, bottomright, bottomleft\n",
-    "m.add_north_arrow(position=\"topright\", scale=1, angle=0)\n",
+    "# Add north-arrow. Placements are: top-right, top-left, bottom-right, bottom-left\n",
```

Comment:
Might be nice to provide example of using relative location here rather than named locations as it's the more flexible approach.

Reply:
I'll make a task to add that, thanks!